wdl-engine/src/eval/v1/workflow.rs

```diff
@@ -69,12 +71,48 @@ use crate::ScopeRef;
 use crate::TaskExecutionBackend;
 use crate::Value;
 use crate::WorkflowInputs;
+use crate::config::Config;
 use crate::diagnostics::if_conditional_mismatch;
 use crate::diagnostics::output_evaluation_failed;
 use crate::diagnostics::runtime_type_mismatch;
 use crate::v1::ExprEvaluator;
 use crate::v1::TaskEvaluator;
 
+/// Helper for formatting a workflow or task identifier for a call statement.
+fn id(namespace: Option<&str>, target: &str, alias: &str, scatter_index: &str) -> String {
+    if alias != target {
+        match namespace {
+            Some(ns) => {
+                format!(
+                    "{ns}-{target}-{alias}{sep}{scatter_index}",
```

Comment:
I'm uncertain, but perhaps the retry counter should be in here?

Reply:
The retry is internal to the task's evaluation, so not something visible to the workflow call statement site.

Reply:
Okay, that's fine then.